dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -1111,10 +1138,12 @@ Spdp::handle_participant_crypto_tokens(const DDS::Security::ParticipantVolatileM
   // If discovery hasn't initialized / validated this participant yet, ignore volatile message
   DiscoveredParticipantIter iter = participants_.find(src_participant);
   if (iter == participants_.end()) {
-    ACE_DEBUG((LM_WARNING,
-      ACE_TEXT("(%P|%t) Spdp::handle_participant_crypto_tokens() - ")
-      ACE_TEXT("received tokens for undiscovered participant %C. Ignoring.\n"),
-               std::string(DCPS::GuidConverter(src_participant)).c_str()));
+    if (OpenDDS::DCPS::security_debug.auth_warn) {
+      ACE_ERROR((LM_WARNING,
```

Comment:
```suggestion
      ACE_DEBUG((LM_WARNING,
```

Reply:
Is it that this should be optionally compiled because of the possibility of it being run a lot?

Reply:
Is it really an `ACE_ERROR`?  If so, why is the log priority `LM_WARNING`?  Based on the comment it doesn't seem like "an error" - it's possible to get these messages in a different order, that's not a a problem.

Reply:
I haven't been really trying to make the decision of what logging level a message should be in these changes. The reason I changed what log macro is being used is that the ACE book says `ACE_ERROR` is for "errors and warnings", although I see I haven't applied that change consistently.

Reply:
I don't have the ACE book in front of me to get more context on that.  Based on Log_Msg.h/cpp it's just setting the 'operation status' which is data that can be logged.  In a simple model where you make an OS or stdlib call and it returns an error, I guess you can differentiate those easily.  This code is a few more layers up, so it's not as straightforward.
The other consideration (probably more important) is for the reader/maintainer of this code: Do we want to communicate ERROR here?

Reply:
> I don't have the ACE book in front of me to get more context on that. Based on Log_Msg.h/cpp it's just setting the 'operation status' which is data that can be logged.

I knew it set the status, but the ACE book says if `ACE_NDEBUG` is defined it would leave out `ACE_DEBUG` statements, assumingly by using something like `#define ACE_DEBUG(X)`. In fact it states that's the entire reason for `ACE_ERROR` and `ACE_DEBUG` being macros. However you're right it doesn't do that, at least not anymore. I looked at the code and did a test to be sure and yea, `ACE_NDEBUG` doesn't have an effect on `ACE_DEBUG`. I feel betrayed...

Reply:
> The other consideration (probably more important) is for the reader/maintainer of this code: Do we want to communicate ERROR here?

Regardless of what the logging macros are doing I think that should be the log category's job.